### DISCUSSION TITLE
NZ State Migrations - Customer Addresses

### DISCUSSION THREAD

**Discussion by pasdo501** [woocommerce/woocommerce#39503] [General]
Hi there,

I'm not sure if this is the right place to raise this or not, so apologies if not. [As part of v7.2.1, new codes for 'states' in NZ were updated](https://github.com/woocommerce/woocommerce/releases/tag/7.2.1). While the DB migration that runs as part of this release updates the codes in a bunch of places, one place it does not update is the usermeta table - specifically billing_state / shipping_state. I couldn't find any discussion on this online, but again apologies if this has already been discussed somewhere.

This results in any existing customer's address seeing a blank for the 'state' since it doesn't match the new code anymore. The site I was updating was using the region for shipping purposes, so it meant existing customers would have to enter their address again (although in this case there was a 'fall through' shipping method that ended up having this cause other issues but that's specific to this particular site.

I'm just wondering if not updating the customer's address information was done on purpose or if it was just missed? I suspect for NZ it poses less of a problem given that the 'state' field is optional by default, although if such an update was done in the future for a country where that's not the case it may be a bit more annoying. It wouldn't be critical, but I can imagine it could lead to some conversion loss & don't know if it may also cause other issues, for with subscriptions (pure speculation).

**Comment by jorgeatorres** [comment]
Hi @pasdo501,

This seems like a valid problem. [I've opened an issue in our repo](https://github.com/woocommerce/woocommerce/issues/39676) so that we can take a look at fixing it. Feel free to follow along to track progress.
